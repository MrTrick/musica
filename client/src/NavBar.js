import React from 'react';
import { withStyles } from '@material-ui/core/styles';

import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';

import Logo from './Logo';
import InputSearch from './InputSearch';

const styles = (theme) => ({
  grow: {
    flexGrow: 1,
  },
});

function NavBar(props) {
  const {classes} = props;

  return (<AppBar position="static">
    <Toolbar>
      <Logo/>
      <div className={classes.grow} />
      <InputSearch />
    </Toolbar>
  </AppBar>);
}

NavBar.propTypes = {};
//TODO: Add handler definitions, etc!

export default withStyles(styles)(NavBar);
