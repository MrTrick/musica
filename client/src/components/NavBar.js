import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';

import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';

import Logo from './Logo';
import InputSearch from './InputSearch';

const styles = (theme) => ({
  grow: {
    flexGrow: 1,
  },
});

function NavBar(props) {
  const {classes, handleSearch, filter} = props;

  return (<AppBar position="static">
    <Toolbar>
      <Logo/>
      <div className={classes.grow} />
      <InputSearch
        value={filter}
        handleChange={handleSearch}
      />
    </Toolbar>
  </AppBar>);
}

NavBar.propTypes = {
  filter: PropTypes.string.isRequired,
  handleSearch: PropTypes.func.isRequired
};


export default withStyles(styles)(NavBar);
