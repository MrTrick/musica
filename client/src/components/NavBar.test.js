import React from 'react';
import ReactDOM from 'react-dom';
import NavBar from './NavBar';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <NavBar filter='' handleSearch={()=>{}} />,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('looks like it should', () => {
  const wrapper = shallow(
    <NavBar filter='Alt Rock' handleSearch={()=>{}} />
  ).dive(); //Go down one level because top-level isn't helpful.

  expect(wrapper).toMatchSnapshot();
});
