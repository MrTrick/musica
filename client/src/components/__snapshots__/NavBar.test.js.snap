// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`looks like it should 1`] = `
<WithStyles(AppBar)
  position="static"
>
  <WithStyles(Toolbar)>
    <Logo />
    <div
      className="NavBar-grow-73"
    />
    <WithStyles(InputSearch)
      handleChange={[Function]}
      value="Alt Rock"
    />
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
